python_modules/dagster/dagster_tests/general_tests/utils_tests/test_cached_method.py

```diff
@@ -131,3 +133,36 @@ def a_method(self, arg1: str, arg2: int) -> str:
     # only one entry
     assert len(obj.__dict__) == 1
     assert len(obj.__dict__[CACHED_METHOD_CACHE_FIELD][MyClass.a_method.__name__]) == 1
+
+
+def test_async_cached_method() -> None:
+    class MyClass:
+        def __init__(self, attr1) -> None:
+            self._attr1 = attr1
+            self.calls = []
+
+        @cached_method
+        async def my_method(self, arg1) -> Tuple:
+            self.calls.append(arg1)
+            await asyncio.sleep(0.25 * random.random())
+            return (arg1, self._attr1)
+
+    obj1 = MyClass(4)
+    assert obj1.calls == []
+    assert asyncio.run(obj1.my_method(arg1="a")) == ("a", 4)
+    assert obj1.calls == ["a"]
+    assert asyncio.run(obj1.my_method(arg1="a")) == ("a", 4)
+    assert asyncio.run(obj1.my_method(arg1="b")) == ("b", 4)
+    assert asyncio.run(obj1.my_method(arg1="a")) == ("a", 4)
+    assert obj1.calls == ["a", "b"]
+
+    async def run_my_method_a_bunch() -> List[Tuple]:
+        return await asyncio.gather(*[obj1.my_method(arg1="a") for i in range(100)])
+
+    assert asyncio.run(run_my_method_a_bunch()) == [("a", 4)] * 100
+    assert obj1.calls == ["a", "b"]
+
+    obj2 = MyClass(5)
+    assert asyncio.run(obj2.my_method(arg1="a")) == ("a", 5)
+    assert asyncio.run(obj2.my_method(arg1="b")) == ("b", 5)
+    assert obj2.calls == ["a", "b"]
```

Comment:
you can use `is` checks for ensuring that you are getting the same in memory object back